src/projection.jl

```diff
@@ -167,7 +167,7 @@ end
 
 # For arrays of numbers, just store one projector:
 function ProjectTo(x::AbstractArray{T}) where {T<:Number}
-    element = ProjectTo(zero(T))
+    element = T <: Irrational ? ProjectTo{Real}() : ProjectTo(zero(T))
```

Comment:
Maybe:

```suggestion
    element = ProjectTo(float(zero(T)))
```

but that is bad for Rational

So we could instead introduce like was discussed before:
`continuous` (would `compact` be more correct)?

which would be
```
continuous(x) = float(x)
continuous(x::Rational) = x
```


```suggestion
    element = ProjectTo(continuous(zero(T)))
```

Reply:
My other hack was to write `promote_type(typeof(1//2), Bool)` which preserves Rational but promotes Irrational to Float64.

Reply:
If there's to be a special function like `continuous` which acts on types, should it immediately return the projector? Could be `ProjectTo(::Type{<:Number})` or perhaps that's too weird if we may later want Type to be non-diff... but some `projectto_from_type`?

Reply:
If you think that is best.
I would still want to wrap that in a function with a clearer name.

Reply:
That does start to feel like an API decision, and my vote here is to kick the can down the road, fix Irrational for now & see if any other need turns up later.

Reply:
I trust your judgement here